README.md

```diff
@@ -1542,6 +1546,123 @@ myVar is evaluated in a boolean context.
 - [Falsy (MDN)](https://developer.mozilla.org/en-US/docs/Glossary/Falsy)
 - [Truthy and Falsy values in JS - Josh Clanton](http://adripofjavascript.com/blog/drips/truthy-and-falsy-values-in-javascript.html)
 
+### Anamorphisms and Catamorphisms
+
+#### Anamorphisms
+
+Anamorphisms are functions that map from some object to a more complex structure containing the type of the object. It is the process of *unfolding* a simple structure into a more complex one. Consider unfolding an integer to a list of integers. The integer is our initial object and the list of integers is the more complex structure.
+
+**Sample code**
+
+```js
+function downToOne(n) {
+  const list = [];
+
+  for (let i = n; i > 0; --i) {
+    list.push(i);
+  }
+
+  return list;
+}
+
+downToOne(5)
+  //=> [ 5, 4, 3, 2, 1 ]
+```
+
+#### Catamorphisms
+
+Catamorphisms are the opposite of Anamorphisms, in that they take objects of more complex structure and *fold* them into simpler structures. Take the following example `product` which take a list of integers and returns a single integer.
+
+**Sample code**
+
+```js
+function product(list) {
+  let product = 1;
+
+  for (const n of list) {
+    product *= n;
+  }
+
+  return product;
+}
+
+product(downToOne(5)) // 120
+```
+
```

Comment:
Missing external resources section

Reply:
Added the external resources subsection for Anamorphisms and Catamorphisms.